### PR TITLE
Deprecate AbstractPlatform::getVarcharTypeDeclarationSQL()

### PR DESCRIPTION
**Pull request by morozov**
The method will be renamed in 4.0 (https://github.com/doctrine/dbal/pull/3586).

### CODE REVIEW

**Comment by morozov** [comment]
The AppVeyor build failure should be fixed by https://github.com/doctrine/dbal/pull/5374.

File: src/Platforms/AbstractPlatform.php [hunk 1]
```diff
@@ -220,6 +222,18 @@ public function getVarcharTypeDeclarationSQL(array $column)
...
     }
 
+    /**
+     * Returns the SQL snippet used to declare a string column type.
+     *
+     * @param mixed[] $column
+     *
+     * @return string
+     */
+    public function getStringTypeDeclarationSQL(array $column)
```
Comment (greg0ire) [inline src/Platforms/AbstractPlatform.php:232]: I think it's OK to use a native return type declaration rightaway, isn't it?

```suggestion
     */
    public function getStringTypeDeclarationSQL(array $column): string
```

Comment (morozov) [reply src/Platforms/AbstractPlatform.php:232]: It proxies the method that doesn't have a return type hint, so I'd leave it as is.

Comment (greg0ire) [reply src/Platforms/AbstractPlatform.php:232]: The proxied method is supposed to return a string. If it does not, don't we want to learn about that as soon as possible? We will either have a bug to fix or  better knowledge about the actual return type that should be documented on the proxied method.

Comment (morozov) [reply src/Platforms/AbstractPlatform.php:232]: This way of thinking could be applied to any method that has its return type documented but not enforced. In this specific case, the code returning a non-string may be implemented in a 3rd-party platform. I don't believe this change should be made as part of deprecation.

Comment (greg0ire) [reply src/Platforms/AbstractPlatform.php:232]: Ok then

**Review by greg0ire — APPROVED** [review]

**Pull request merged**